src/python/espressomd/interactions.pyx

```diff
@@ -205,6 +205,85 @@ cdef class NonBondedInteraction:
         raise Exception(
             "Subclasses of NonBondedInteraction must define the required_keys() method.")
 
+
+class NewNonBondedInteraction(ScriptInterfaceHelper):
```

Comment:
Well, it wasn't used before because everything was in Cython, and abstract classes don't work out-of-the-box in Cython:

```
Error compiling Cython file:
------------------------------------------------------------
...

import abc
cdef class HydrodynamicInteraction(FluidActor, metaclass=abc.ABCMeta):
                                                                    ^
------------------------------------------------------------
/work/jgrad/es/src/python/espressomd/lb.pyx:172:69: C classes cannot take keyword bases.
```

or

```
Error compiling Cython file:
------------------------------------------------------------
...

import abc
cdef class FluidActor(abc.ABC):
                        ^
------------------------------------------------------------
/work/jgrad/es/src/python/espressomd/lb.pyx:34:25: First base of 'FluidActor' is not an extension type
```

The usual workaround was to declare a python abstract class and use it to register the cdef class to make it abstract, which is less elegant and error-prone. Now that the module is in Python, we can use ABC metaclasses.